File: src/Entity/Materiel.php
<?php

namespace App\Entity;

use App\Repository\MaterielRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass: MaterielRepository::class)]
class Materiel
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $idMateriel = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: "Libelle ne doit pas etre vide")]
    #[Assert\Length(min:8,minMessage:"Libelle doit au moins contenir 8 caracteres")]
    private ?string $Libelle = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: "S'il vous plait ecrire une description")]
    #[Assert\Length(min:12,minMessage:"Vous devez écrire un texte")]
    private ?string $Description = null;

    #[ORM\Column]
    private ?int $Disponibilite = null;

    #[ORM\Column(length: 255)]
    #[Assert\Image(mimeTypes: ["image/jpeg", "image/png"], mimeTypesMessage: "S'il vous plait télecharger une image valide JPEG ou PNG")]
    private ?string $Image = null;

    #[ORM\Column]
    #[Assert\NotBlank(message: "Le prix est important")]
    #[Assert\GreaterThan(value: 0, message: "Prix doit etre supérieur à 0")]
    private ?float $Prix = null;

    #[ORM\ManyToOne(inversedBy: 'materiels')]
    #[ORM\JoinColumn(name: 'id_categorie', referencedColumnName: 'id_categorie')]
    private ?Categorie $idCategorie = null;

    #[ORM\ManyToOne(inversedBy: 'materiels')]
    private ?User1 $IdUser = null;

    public function getidMateriel(): ?int
    {
        return $this->idMateriel;
    }

    public function getLibelle(): ?string
    {
        return $this->Libelle;
    }

    public function setLibelle(string $Libelle): static
    {
        $this->Libelle = $Libelle;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->Description;
    }

    public function setDescription(string $Description): static
    {
        $this->Description = $Description;

        return $this;
    }

    public function getDisponibilite(): ?int
    {
        return $this->Disponibilite;
    }

    public function setDisponibilite(int $Disponibilite): static
    {
        $this->Disponibilite = $Disponibilite;

        return $this;
    }

    public function getImage(): ?string
    {
        return $this->Image;
    }

    public function setImage(?string $Image): self
    {
        $this->Image = $Image;

        return $this;
    }

    public function getPrix(): ?float
    {
        return $this->Prix;
    }

    public function setPrix(float $Prix): static
    {
        $this->Prix = $Prix;

        return $this;
    }

    public function getIdCategorie(): ?Categorie
    {
        return $this->idCategorie;
    }

    public function setIdCategorie(?Categorie $idCategorie): static
    {
        $this->idCategorie = $idCategorie;

        return $this;
    }

    public function getIdUser(): ?User1
    {
        return $this->IdUser;
    }

    public function setIdUser(?User1 $IdUser): static
    {
        $this->IdUser = $IdUser;

        return $this;
    }

    public function getCategorieLibelle(): ?string
    {
    return $this->idCategorie ? $this->idCategorie->getLibelleCategorie() : null;
    }

    
}

File: src/main/java/Entity/Materiel.java
package Entity;

public class Materiel {
    private int id;

    private String LibelleMateriel;

    private String Description;

    private int Disponibilite;

    private String ImageMateriel;


    private double Prix;
    private int id_categorie;

    private int id_user;


    public Materiel()
    {

    }

    public int getId_user() {
        return id_user;
    }

    public Materiel(String libelleMateriel, String description, int disponibilite, String imageMateriel, double prix, int id_categorie, int id_user) {
        LibelleMateriel = libelleMateriel;
        Description = description;
        Disponibilite = disponibilite;
        ImageMateriel = imageMateriel;
        Prix = prix;
        this.id_categorie = id_categorie;
        this.id_user = id_user;
    }

    public Materiel(int id, String libelleMateriel, String description, int disponibilite, String imageMateriel, double prix, int id_categorie) {
        this.id = id;
        LibelleMateriel = libelleMateriel;
        Description = description;
        Disponibilite = disponibilite;
        ImageMateriel = imageMateriel;
        Prix = prix;
        this.id_categorie = id_categorie;
    }


    public Materiel(int id, String libelleMateriel, String description, int disponibilite, String imageMateriel, double prix) {
        this.id = id;
        LibelleMateriel = libelleMateriel;
        Description = description;
        Disponibilite = disponibilite;
        ImageMateriel = imageMateriel;
        Prix = prix;
    }


    public Materiel(String libelleMateriel, String description, int disponibilite, String imageMateriel, double prix, int id_categorie) {
        LibelleMateriel = libelleMateriel;
        Description = description;
        Disponibilite = disponibilite;
        ImageMateriel = imageMateriel;
        Prix = prix;
        this.id_categorie = id_categorie;
    }



    public Materiel(String libelleMateriel, String description, int disponibilite, String imageMateriel, double prix) {
        LibelleMateriel = libelleMateriel;
        Description = description;
        Disponibilite = disponibilite;
        ImageMateriel = imageMateriel;
        Prix = prix;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getLibelleMateriel() {
        return LibelleMateriel;
    }

    public void setLibelleMateriel(String libelleMateriel) {
        LibelleMateriel = libelleMateriel;
    }

    public String getDescription() {
        return Description;
    }

    public void setDescription(String description) {
        Description = description;
    }

    public int getDisponibilite() {
        return Disponibilite;
    }

    public void setDisponibilite(int disponibilite) {
        Disponibilite = disponibilite;
    }

    public String getImageMateriel() {
        return ImageMateriel;
    }

    public void setImageMateriel(String imageMateriel) {
        ImageMateriel = imageMateriel;
    }

    public double getPrix() {
        return Prix;
    }

    public void setPrix(double prix) {
        Prix = prix;
    }

    public int getId_categorie() {
        return id_categorie;
    }

    public void setId_categorie(int id_categorie) {
        this.id_categorie = id_categorie;
    }

    @Override
    public String toString() {
        return "Materiel{" +
                "id=" + id +
                ", LibelleMateriel='" + LibelleMateriel + '\'' +
                ", Description='" + Description + '\'' +
                ", Disponibilite=" + Disponibilite +
                ", ImageMateriel='" + ImageMateriel + '\'' +
                ", Prix=" + Prix +
                ", id_categorie=" + id_categorie +
                '}';
    }
}